boot/freeldr/freeldr/arch/uefi/uefisetup.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * PROJECT:     Freeldr UEFI Extension
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     UEFI Mach Setup
+ * COPYRIGHT:   Copyright 2022 Justin Miller <justinmiller100@gmail.com>
+ */
+
+#include <uefildr.h>
+
+#include <debug.h>
+DBG_DEFAULT_CHANNEL(WARNING);
+
+/* GLOBALS ********************************************************************/
+
+extern EFI_SYSTEM_TABLE* GlobalSystemTable;
+extern EFI_HANDLE GlobalImageHandle;
+BOOLEAN AcpiPresent = FALSE;
+
+/* FUNCTIONS ******************************************************************/
+
+EFI_STATUS
+UefiHandleService(EFI_STATUS Status)
+{
+    if (Status != EFI_SUCCESS)
+    {
+        TRACE("UEFI Services call has failed with status: %X\n", Status);
+    }
+
+    return Status;
+}
+
+VOID
+MachInit(const char *CmdLine)
+{
+    EFI_STATUS Status = 0;
+    Status = GlobalSystemTable->BootServices->AllocatePool(EfiLoaderData, sizeof(MachVtbl), (void**)&MachVtbl);
+    if (UefiHandleService(Status) != EFI_SUCCESS)
+        ERR("UefiMachInit: Could not allocated memory for MachVtbl");
+
+    /* Setup GOP */
+    UefiInitalizeVideo();
```

Comment:
Move the gop allocation after all the table pointers are initialized, to follow what's done e.g. for xbox.